Added configuration to set service type for Hubble UI (#12929)
cilium-team/cilium: Documentation/gettingstarted/hubble-enable.rst

              --set global.hubble.relay.enabled=true \\
              --set global.hubble.ui.enabled=true

        .. note::

           For cloud environment you may need to set Hubble-UI K8s service type to something other than default ClusterIP.
           You may add param:

> 

Reviewer: I like the change of making the service type configurable, but I do not like that this note sounds like we are recommending this practice for deployments.

Hubble UI gives you access to potentially sensitive data (network traffic and deployed applications), thus I feel uneasy that this sounds like it recommends people expose Hubble UI directly via NodePort or LoadBalancer. I think the proper way to expose Hubble UI is via a HTTP reverse proxy (e.g. using a K8s Ingress), with the proper access protection (such as e.g. TLS + Basic Auth) in place.

I would prefer it if we removed this note here. Again, people who know what they are doing might have legitimate reasons to expose Hubble UI via NodePort, but I do not think it should be a recommendation from our side to do it.

Author: Well, I'm a bit confused.

Just using node port in clouds will auto make service exposed? Can you explain scenario for cloud like AWS?

This just enables us to use nodePort (which is used by different components as well) and without proper reverse proxy (as you mentioned) or 'almost the same' aws-alb-controller + creating K8S Ingress resource this shouldn't be exposed automatically.

So after using this param you need to:
1. install controller... `nginx-ingress` or `AWS-ALB-controller` 
2. Create Ingress resource.
Both cases support TLS and Basic Auth (tbh useless and I wouldn't recommend this). They support OAuth as well.

This sounds to me like 'people will know what they are doing'.

Author: @gandro  I changed documentation, is it better now?

Author: or maybe you can give me a hint how to phrase this. English isn't my native language and I'm not the best guy when it comes to explanations.

Reviewer 2: it's not native either for @gandro, or indeed for the majority of cilium ;)

Reviewer: @wizard580: You are correct that using NodePort will probably be fine in most cloud setups, as the node network is not accessible to the internet. I could have formulated that better, I was mostly worried about LoadBalancer, as an unexperienced user might not understand that this would expose it to the Internet on most managed K8s solutions (e.g. GKE/AKS).

In any case, the new message gives a lot more context of how to expose it properly, I think this will help users to make an informed choice. Thanks!

              --set global.hubble.relay.enabled=true \\
              --set global.hubble.ui.enabled=true

        .. note::

           For some environments, if you want to expose Hubble-UI to your protected network - 

>            we may recommend you to use reverse proxy with TLS and OAuth enabled. 

Reviewer 3: nit: can you add OAuth and UI into `spelling_wordlist.txt` by running `Documentation/update-spelling_wordlist.sh  OAuth UI` command.

           Ex: nginx-ingress-controller or aws-alb-ingress-controller
           

>            For this purpose you may need to set Hubble-UI K8s service type to something other than default ClusterIP.
>            For this you can use param:
> 

Reviewer 3: To avoid duplicate `for this`, and in other places, we refer `helm option` instead of param.

```suggestion
           For this purpose you may need to set Hubble UI K8s service type to something other than default ClusterIP with ``helm`` option `hubble-ui.service.type`
```

           .. parsed-literal::

              --set hubble-ui.service.type=NodePort

           Be aware that Hubble-UI gives access to potentially sensitive information about running services in your cluster

>            so be sure that you use properly configured TLS and Authentification to prevent security incidents.

Reviewer 3: Authentification is typo for Authentication. Authorization could be mentioned as well.
